pytest_lab/locker.py

```diff
@@ -122,22 +122,31 @@ def __init__(self, config, backend, ttl=30):
         self._thread = None
         self._stop = threading.Event()
 
-    def aquire(self, key, user=None):
+    def aquire(self, key, wait_on_locks, user=None):
         record = self.backend.read(key)
 
-        if record and record.ttl:
-            logger.error('{} is locked by {}, waiting {} seconds for lock '
-                         'to expire...'.format(key, record.value, record.ttl + 1))
-            start = time.time()
-            while time.time() - start < record.ttl + 1:
-                record = self.backend.read(key)
-                if not record:
+        def check_lock(record):
+            if record and record.ttl:
+                logger.error(
+                    '{} is locked by {}, waiting {} seconds for lock '
+                    'to expire...'.format(key, record.value, record.ttl + 1))
+                start = time.time()
+                while time.time() - start < record.ttl + 1:
+                    record = self.backend.read(key)
+                    if not record:
+                        return True
+                    time.sleep(0.5)
```

Comment:
You don't have to poll this aggressively. You can sleep for half of ttl's length.

Reply:
Polling every half a second is from previous code. 
I agree, half of ttl should be frequent enough in this scenario.

Reply:
I looked it up, you're right. Tyler made the change.